clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/record/ByteBufferValueRecord.java

```diff
@@ -0,0 +1,27 @@
+package com.linkedin.davinci.store.record;
+
+/**
+ * This class encapsulates a value from venice storage accompanied by the schema
+ * id that was used to serialize the value.
+ *
+ * TODO: This class should probably be superseded by {@link com.linkedin.davinci.store.record.ValueRecord}. Unfortunately,
+ * MANY interfaces in the ingestion path rely on the Bytebuffer interface, where ValueRecord relies on ByteBuf. Until
+ * we rectify that, this is our stand in.
+ */
+public class ByteBufferValueRecord<T> {
+  T value;
+  int writerSchemaId = -1;
```

Comment:
Can we make these properties final?

Reply:
Sure can.  Let me do that.

Reply:
Done, and intellij prompted me to convert the class to a record, so I did.

Reply:
and promptly put it back to class. Boo